react-hyf/react-hyf-client/src/ProfileGenerator.js

```diff
@@ -0,0 +1,69 @@
+import React, { Component } from 'react';
+
+class ProfileGenerator extends Component {
+  constructor() {
+    super();
+    this.state = {
+      error: null,
+      isLoaded: false,
+      people: [],
+      randomPerson: '',
+      gender: 'random',
+    };
+  }
+
+  componentDidMount() {
+    fetch('https://uinames.com/api/?amount=10&ext')
+      .then(response => response.json())
+      .then(data => {
+        this.setState({
+          isLoaded: true,
+          people: data,
+        });
+      })
+      .catch(error => {
+        this.setState({
+          isLoaded: true,
+          error,
+        });
+      });
+  }
+
+  handleSubmit = event => {
+    event.preventDefault();
+    const randNum = Math.floor(Math.random() * this.state.people.length);
+    const randPerson = this.state.people[randNum];
+    this.setState({ randomPerson: randPerson });
+  };
+
+  render() {
+    const { error, isLoaded } = this.state;
+
+    return (
+      <div>
+        {error ? (
+          <div className="attention">Error: {error.message}</div>
+        ) : (
+          isLoaded && <div className="attention">Loading...</div> && (
+            <form className="profile-form" onSubmit={this.handleSubmit} onClick={this.handleSubmit}>
```

Comment:
An `onClick` on a form is a bit strange, either a button gets clicked, or a form gets submitted.

Reply:
onSubmit eventListener transformed to Space button press